plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -234,7 +234,10 @@ def process_main(self):
                 if content_type == man_list:
                     self._process_manifest_list(manifest, digest, available_blobs, tag)
                 else:
-                    self._process_manifest(manifest, digest, available_blobs, tag)
+                    foreign_layers = self._process_manifest(manifest, digest, available_blobs, tag)
+                    if foreign_layers:
+                        # we don't want to process schema1 manifest with foreign layers
```

Comment:
so we can figure out which layer type is, just on schema2, while processing schema2 if we stumble across foreign layer, we skip its download( by looking at the mediatype), unfortunately we cannot do the same with schema1 which contains foreign layer, because during the conversion on the fly of schema2-->schema1, manifest schema1 contains just the list of sha blobs and you cannot distinguish which on one is ok and which one is normal.
Because of that we decided not to download schema1 manifests that have foreign layers.
In addition docker pull of schema1( with older client) would not work anyway.